pythran/pythonic/bisect/bisect_impl.hpp

```diff
@@ -0,0 +1,25 @@
+#ifndef PYTHONIC_BISECT_BISECT_IMPL_HPP
+#define PYTHONIC_BISECT_BISECT_IMPL_HPP
+
+#include "pythonic/utils/proxy.hpp"
+#include "pythonic/bisect/bisect.hpp"
+
+#include <algorithm>
+
+namespace pythonic {
+
+    namespace bisect {
+        template <class X, class A>
+            size_t bisect(X const& x, A const& a, long lo, long hi) {
+                if(hi<0) hi = x.end() - x.begin();
+                return std::upper_bound(x.begin()+lo, x.begin()+hi, a) - x.begin();
+            }
+
+        PROXY_IMPL(pythonic::bisect, bisect);
+
+    }
+
+}
+
+#endif
+
```

Comment:
On Mon, Mar 23, 2015 at 01:37:56AM -0700, pbrunet wrote:
> > +
> > +    namespace bisect {
> > +        template <class X, class A>
> > +            size_t bisect(X const& x, A const& a, long lo, long hi) {
> > +                if(hi<0) hi = x.end() - x.begin();
> > +                return std::upper_bound(x.begin()+lo, x.begin()+hi, a) - x.begin();
> > +            }
> > +
> > +        PROXY_IMPL(pythonic::bisect, bisect);
> > +
> > +    }
> > +
> > +}
> > +
> > +#endif
> > +
> 
> I choose this to break dependencies between files and classes.
> Usually, we create .hpp and .cpp files in a c++ project but it is not done here because we want to keep inlining as much as possible. I don't see any other solution to both keep inlining and correct file dependencies like in other c++ project.

There 's something I'm missing here: in that case you should have the
impl include the header and not the following, so that we can include
the header without getting the impl?